tests/test_regex.py

```diff
@@ -74,5 +300,37 @@ def test_named_groups(self):
     def test_begining(self):
         assert_can_generate('^abc')
 
+    def test_caret_in_the_middle_does_not_generate_anything(self):
+        r = re.compile('a^b')
+
+        with pytest.raises(he.NoSuchExample):
+            h.find(regex(r), r.match)
```

Comment:
1. `'a^b'` does not match any string.  Thanks to the filter, the strategy therefore generates no examples.  (so far, so goo)
2. Using `r.match` as the predicate is therefore redundant.  
3. `lambda s: True` would be more efficient, and clearly does not exclude possible outputs.
4. Using `regex(r).example()` is more efficient again - it basically uses `find` internally and stops at the first example, where `find` keeps trying until it finds the simplest example.